app/models/manageiq/providers/amazon/inventory/parser/cloud_manager.rb

```diff
@@ -191,11 +191,49 @@ def instances
         ),
       )
 
-      instance_hardware(persister_instance, instance, flavor)
+      @tag_mapper = ContainerLabelTagMapping.mapper
+
+      labels = parse_labels(instance["tags"])
+      taggings = map_labels_to_tags("Vm", labels)
+      map_taggings(persister_instance, taggings)
       vm_and_template_labels(persister_instance, instance["tags"] || [])
     end
   end
 
+  def parse_labels(entity)
+    parse_identifying_attributes(entity, 'labels')
+  end
+
+  def parse_identifying_attributes(attributes, section, source = "amazon")
+    result = []
+    return result if attributes.nil?
+      attributes.each do |tag|
+        custom_attr = {
+          :section => section,
+          :name    => tag["key"],
+          :value   => tag["value"],
+          :source  => source
```

Comment:
If you *only* use this function to feed map_labels, you only need :name and :value.

Reply:
Ok thanks!